pkg/gatewayserver/io/io.go

```diff
@@ -159,6 +159,11 @@ func (c *Connection) HandleUp(up *ttnpb.UplinkMessage) error {
 		}
 		md.UplinkToken = buf
 		md.DownlinkPathConstraint = c.gateway.DownlinkPathConstraint
+
+		if len(c.gateway.Antennas) > int(md.AntennaIndex) {
```

Comment:
Where does this come from? 

Should we be getting `antennnas` here?

https://github.com/TheThingsNetwork/lorawan-stack/blob/96ebc355c10a456224f1957bf494c10f1fef38f3/pkg/gatewayserver/gatewayserver.go#L361-L371

Also, make sure that `antennas` is returned when GS uses cluster authentication (i.e. in an unauthenticated gateway context, when the gateway uses UDP for example).

Reply:
Should be OK now